Content.Server/Power/EntitySystems/PowerNetSystem.cs

```diff
@@ -311,13 +316,29 @@ private void UpdatePowerConsumer()
             var enumerator = EntityQueryEnumerator<PowerConsumerComponent>();
             while (enumerator.MoveNext(out var consumer))
             {
+                if (!consumer.Powered)
+                {
+                    // If unpowered, loop through all the wires and check their supply
+                    var nodeComponent= Comp<NodeContainerComponent>(consumer.Owner);
+                    foreach (var nodeValue in nodeComponent.Nodes.Values)
+                    {
+                        var poweredWire = nodeValue.NodeGroup is IBasePowerNet { NetworkNode: { LastCombinedSupply: >0 } };
+                        if (poweredWire)
+                        {
+                            // Set the consumer's voltage to the wire if it has power
+                            consumer.Voltage = (Voltage) nodeValue.NodeGroupID;
+                            break;
+                        }
+                    }
+                }
```

Comment:
Someone more familiar with power should look at this, e.g. pjb or electro maybe.